pytransform3d/transform_manager.pyi

```diff
@@ -1,15 +1,17 @@
+import abc
+
 import numpy as np
 import numpy.typing as npt
 import scipy.sparse as sp
 from mpl_toolkits.mplot3d import Axes3D
 from typing import Dict, Tuple, List, Union, Set, Hashable, Any
 
 
-class TransformManager(object):
+class TransformGraphBase(abc.ABC):
     strict_check: bool
     check: bool
-    transforms: Dict[Tuple[Hashable, Hashable], np.ndarray]
     nodes: List[Hashable]
+    transforms: Dict[Tuple[Hashable, Hashable], np.ndarray]
```

Comment:
This would force the future `TemporalTransformManager` to use `np.ndarray`s to store timeseries of transformation. I rather suggest the following:

```python
from typing import Protocol

class Transform(Protocol):

    def get_matrix(self, time: Optional[float]):
        ...
```

```suggestion
    transforms: Dict[Tuple[Hashable, Hashable], Transform]
```

Reply:
This is a property now